framework/src/geomsearch/NearestNodeLocator.C

```diff
@@ -212,6 +223,82 @@ NearestNodeLocator::nearestNode(dof_id_type node_id)
   return _nearest_node_info[node_id]._nearest_node;
 }
 
+void
+NearestNodeLocator::updatePatch(std::vector<dof_id_type> & recheck_slave_nodes)
+{
+  Moose::perf_log.push("NearestNodeLocator::updatePatch()", "Execution");
+
+  std::vector<dof_id_type> trial_master_nodes;
+
+  // Build a bounding box.  No reason to consider nodes outside of our inflated BB
+  UniquePtr<BoundingBox> my_inflated_box = NULL;
+
+  const std::vector<Real> & inflation = _mesh.getGhostedBoundaryInflation();
+
+  // This means there was a user specified inflation... so we can build a BB
+  if (inflation.size() > 0)
+  {
+    BoundingBox my_box = MeshTools::create_local_bounding_box(_mesh);
+
+    Point distance;
+    for (unsigned int i = 0; i < inflation.size(); ++i)
+      distance(i) = inflation[i];
+
+    my_inflated_box =
+        libmesh_make_unique<BoundingBox>(my_box.first - distance, my_box.second + distance);
+  }
+
+  // Data structures to hold the Nodal Boundary conditions
+  ConstBndNodeRange & bnd_nodes = *_mesh.getBoundaryNodeRange();
+  for (const auto & bnode : bnd_nodes)
+  {
+    BoundaryID boundary_id = bnode->_bnd_id;
+    dof_id_type node_id = bnode->_node->id();
+
+    // If we have a BB only consider saving this node if it's in our inflated BB
+    if (!my_inflated_box || (my_inflated_box->contains_point(*bnode->_node)))
+    {
+      if (boundary_id == _boundary1)
+        trial_master_nodes.push_back(node_id);
+    }
+  }
+
+  // Convert trial master nodes to a vector of Points. This would be used to construct the KDTree.
+  std::vector<Point> master_points(trial_master_nodes.size());
+  for (unsigned int i = 0; i < trial_master_nodes.size(); ++i)
+  {
+    const Node & node = _mesh.nodeRef(trial_master_nodes[i]);
+    master_points[i] = node;
+  }
+
+  // Create object kd_tree of class KDTree using the coordinates of trial
+  // master nodes.
+  KDTree kd_tree(master_points, _mesh.getMaxLeafSize());
+
+  NodeIdRange trial_slave_node_range(recheck_slave_nodes.begin(), recheck_slave_nodes.end(), 1);
```

Comment:
Would `recheck_slave_node_range` be a better name?

Reply:
Yes, I guess recheck_slave_node_range would be better name to avoid confusion.